### PR TITLE
Document test discovery naming conventions

### PR DESCRIPTION
**Pull request by wholtz**
### Checklist
- [ X] Pull request details were added to HISTORY.rst

### CODE REVIEW

File: docs/writing_tests.rst [hunk 1]
```diff
@@ -160,7 +160,11 @@ Multiple workflows can use the same custom test like this:
...
 
+Custom tests must follow the `conventions for Python test discovery
+<https://docs.pytest.org/en/7.2.x/explanation/goodpractices.html#conventions-for-python-test-discovery>`_,
+which constrains the names of files and functions containing custom tests.
+
 .. note::
 
     stdout and stderr are available as files in the root of the
-    ``workflow_dir`` as ``log.out`` and ``log.err`` respectively.
+    ``workflow_dir`` as ``log.out`` and ``log.err`` respectively.
```
Comment (rhpvorderman) [inline docs/writing_tests.rst:170]: I presume this is a ``git`` error as the line has not changed? Or am I missing something?

File: docs/writing_tests.rst [hunk 2]
```diff
@@ -160,7 +160,11 @@ Multiple workflows can use the same custom test like this:
 points to the folder where the named workflow was executed. This allows writing
 of advanced python tests for each file produced by the workflow.
 
+Custom tests must follow the `conventions for Python test discovery
+<https://docs.pytest.org/en/7.2.x/explanation/goodpractices.html#conventions-for-python-test-discovery>`_,
```
Comment (rhpvorderman) [inline docs/writing_tests.rst:164]: This statically codes the link to the 7.2.x documentation. Please find the link for the dynamically updating lastest stable version of pytest.

**Review by rhpvorderman — CHANGES_REQUESTED** [review]

**Comment by rhpvorderman** [comment]
Thanks. Just one minor comment about the link itself.

File: docs/writing_tests.rst (hunk 1 above)
Comment (wholtz) [reply docs/writing_tests.rst]: The end of file new line was unintentionally deleted. I have put it back.

File: docs/writing_tests.rst (hunk 2 above)
Comment (wholtz) [reply docs/writing_tests.rst]: Done.

**Review by rhpvorderman — APPROVED** [review]

**Comment by rhpvorderman** [comment]
Thanks!

**Pull request merged**